fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/IndexKeyValueToPartialRecord.java

```diff
@@ -275,7 +276,11 @@ public boolean copy(@Nonnull Descriptors.Descriptor recordDescriptor, @Nonnull M
             }
             switch (fieldDescriptor.getType()) {
                 case INT32:
-                    value = ((Long)value).intValue();
+                    if (value.getClass().isArray()) {
```

Comment:
I think we want to return the bitmap as a bite array, i.e. in the way this is delivered from the index entry.